### PR TITLE
[FLINK-26993][tests] Wait until checkpoint was actually triggered

### PR DESCRIPTION
**Pull request by zentol**
This is a rather crude fix :/

### CODE REVIEW

File: flink-runtime/src/test/java/org/apache/flink/runtime/checkpoint/CheckpointCoordinatorTest.java [hunk 1]
```diff
@@ -388,8 +388,9 @@ public void testMinCheckpointPause() throws Exception {
             coordinator.triggerCheckpoint(
                     true); // enqueue and later see if it gets executed in the middle of
             // receiveAcknowledgeMessage
-            while (coordinator.getNumberOfPendingCheckpoints()
-                    == 0) { // wait for at least 1 request to be fully processed
+            while (coordinator.getPendingCheckpoints().values().stream()
+                    .noneMatch(pc -> pc.getCheckpointStorageLocation() != null)) {
+                // wait for at least 1 request to be fully processed
```
Comment (pnowojski) [inline flink-runtime/src/test/java/org/apache/flink/runtime/checkpoint/CheckpointCoordinatorTest.java:393]: Can you add a comment above why do we need this hacky check?

**Review by pnowojski — COMMENTED** [review]
Meh, quite hacky solution. +1 for merging it, as a proper solution (for example waiting until `CheckpointCoordinator` actually triggers a checkpoint RPC before acknowledging it) might be non trivial to implement in this test. 

Can you also post an example stack trace in the Jira ticket, so that if someones in the future do a `git blame` trying to understand what's the story behind this check, he can find more info?

File: flink-runtime/src/test/java/org/apache/flink/runtime/checkpoint/CheckpointCoordinatorTest.java (hunk 1 above)
Comment (zentol) [reply flink-runtime/src/test/java/org/apache/flink/runtime/checkpoint/CheckpointCoordinatorTest.java:393]: Done!

**Comment by zentol** [comment]
> Can you also post an example stack trace in the Jira ticket

Done!

**Pull request merged**